js/customize-partial-refresh-settings.js

```diff
@@ -0,0 +1,128 @@
+/* global jQuery, _customizePartialRefreshSettings, JSON, _ */
+/* exported customizePartialRefreshSettings */
+var customizePartialRefreshSettings = ( function( $ ) {
+	'use strict';
+
+	var self, api = wp.customize;
+
+	self = {
+		setting: null
+	};
+
+	$.extend( self, _customizePartialRefreshSettings );
+
+	/**
+	 * Inject the functionality.
+	 */
+	self.init = function() {
+		api.bind( 'ready', function() {
+			self.onChangeSetting();
+		} );
+
+		api.bind( 'setting-partial', function( response ) {
+			var that = $( 'iframe' ).contents(),
+				value = $.trim( response );
+
+			$( self.setting.selector, that ).html( value );
+		} );
+	};
+
+	/**
+	 * Change event for selective settings, requests a new value.
+	 */
+	self.onChangeSetting = function() {
+		$.each( self.settings, function( id, selector ) {
+
+			api( id, function( setting ) {
+
+				setting.bind( function() {
+					self.setting = {
+						id: id,
+						selector: selector
+					};
+					self.request();
+				} );
+			} );
+		} );
+	};
+
+	/**
+	 * Request a new setting value.
+	 *
+	 * @return {jQuery.Deferred}
+	 */
+	self.request = function() {
+		var spinner = $( '#customize-header-actions .spinner' ),
+			active = 'is-active',
+			deferred = $.Deferred(),
+			req = self.debounceRequest();
+
+		spinner.addClass( active );
+
+		req.done( function( response ) {
+			deferred.resolve();
+			api.trigger( 'setting-partial', response );
+			spinner.removeClass( active );
+		} );
+
+		req.fail( function() {
+			deferred.reject.apply( deferred, arguments );
+			api.previewer.refresh();
+			spinner.removeClass( active );
+		} );
+
+		return deferred;
+	};
+
+	/**
+	 * Debounce the requests, allowing setting changes made back-to-back to be sent together.
+	 *
+	 * @return {jQuery.Deferred}
+	 */
+	self.debounceRequest = ( function() {
+		var request, debouncedDeferreds = [];
+
+		request = _.debounce( function() {
+			var req, dirtyCustomized = {};
+
+			api.each( function( value, key ) {
+				if ( value._dirty ) {
```

Comment:
In the Customzier preview, is not `_dirty` undefined since the settings are just `Value` instances and not `Setting` instances. I had to go through some hoops to get the dirty settings to get notified in the preview for Customize REST Resources:

https://github.com/xwp/wp-customize-rest-resources/blob/2e3f11ec7f1fe571e8b926553ddef8d3e2ca8f66/js/rest-resources-preview-manager.js#L49-L56
https://github.com/xwp/wp-customize-rest-resources/blob/2e3f11ec7f1fe571e8b926553ddef8d3e2ca8f66/js/rest-resources-pane-manager.js#L89-L98